python/TestHarness/testers/RunException.py

```diff
@@ -31,7 +31,11 @@ def processResults(self, moose_dir, retcode, options, output):
     # Expected errors and assertions might do a lot of things including crash so we
     # will handle them seperately
     if specs.isValid('expect_err'):
-      if not self.checkOutputForPattern(output, specs['expect_err']):
+      # TODO: specs['redirect_stdout'] is always going to be non-False, so is there a better way?
+      if (options.parallel and options.parallel > 1) and int(specs['max_parallel']) != 1:
```

Comment:
The Testers should all be oblivious to where the output is coming from. RunParallel is going to make the decision about how to collect output. It should just read it all into a single variable and had it back to the Testers.